pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -489,7 +492,7 @@ func (wfs *WorkflowSpec) HasPodSpecPatch() bool {
 // Template is a reusable and composable unit of execution in a workflow
 type Template struct {
 	// Name is the name of the template
-	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
```

Comment:
This field is validated by CLI, UI, and controller.  But the only restriction when the user using swagger.json or SDK they need to set a dummy value on this.